lib/src/json_schema/constants.dart

```diff
@@ -37,6 +37,15 @@
 //     THE SOFTWARE.
 
 class JsonSchemaValidationRegexes {
+  // Spec: the ISO 8601 ABNF as given in Appendix A of RFC 3339
+  // From: https://www.oreilly.com/library/view/regular-expressions-cookbook/9781449327453/ch04s07.html
+  static RegExp duration = RegExp(
+      r'^P(\d{1,}W|T(\d{1,}H(\d{1,}M(\d{1,}S)?)?|\d{1,}M(\d{1,}S)?|\d{1,}S)|(\d{1,}D|\d{1,}M(\d{1,}D)?|\d{1,}Y(\d{1,}M(\d{1,}D)?)?)(T(\d{1,}H(\d{1,}M(\d{1,}S)?)?|\d{1,}M(\d{1,}S)?|\d{1,}S))?)$');
+
+  // Spec: https://datatracker.ietf.org/doc/html/rfc4122
+  static RegExp uuid =
+      RegExp(r'^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-5][0-9a-fA-F]{3}-[089abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$');
```

Comment:
Strangely, some optional spec tests want us to support hypothetical new UUID versions. If we think that's valid, I can make the change and re-enable those tests. It struck me as odd though.